### ISSUE TITLE
Add a drawing tool with saving as new image function

### ISSUE THREAD

**Issue by Clasen00** [openseadragon/openseadragon#2425]
Hello, openseadragon is a great tool and help me a lot! Thank you for your job!

I would like to to add on openseadragon.viewer canvas a tool, which will let me to draw a shape on it and then save it as a new image. Is there is something ready to use for openseadragon with this functional or similar? 

Any other suggestions on how to do it will be great to!

Thank you!

**Comment by pearcetm** [comment]
Are you wanting a new image that includes the original image plus whatever overlay you've drawn onto it, baked into a single file? Or do you want to draw a shape in order to crop the image into a new file? I'm not sure if there are tools out there that do these things already.

**Comment by Clasen00** [comment]
> Are you wanting a new image that includes the original image plus whatever overlay you've drawn onto it, baked into a single file? Or do you want to draw a shape in order to crop the image into a new file? I'm not sure if there are tools out there that do these things already.

@pearcetm I just suggested that if I have a canvas from openseadragon then I could draw on this canvas somehow and get from it a new image with my draw on it

**Comment by pearcetm** [comment]
You can get the canvas with `viewer.drawer.canvas` and grab the context using `canvas.getContext('2d')`, and then draw whatever you'd like onto it before getting the new image with drawing by calling `canvas.getDataURL()`. However, as soon as you move the viewer the drawing will be overwritten by the image again. The best approach will depend on exactly what you envision as your use case.

**Comment by Clasen00** [comment]
> You can get the canvas with `viewer.drawer.canvas` and grab the context using `canvas.getContext('2d')`, and then draw whatever you'd like onto it before getting the new image with drawing by calling `canvas.getDataURL()`. However, as soon as you move the viewer the drawing will be overwritten by the image again. The best approach will depend on exactly what you envision as your use case.

Yes, thank you @pearcetm ! I think it will be bad solution for me. Better algorithm is draw a svg, save it to json with coords and send it into backend. Will try to figure it out somehow like this:)
